framework/src/main/java/org/checkerframework/common/accumulation/AccumulationValue.java

```diff
@@ -0,0 +1,153 @@
+
+  /**
+   * If the underlying type is a type variable or a wildcard, then this is a set of accumulated
+   * values. Otherwise, it is null.
+   */
+  private @Nullable Set<String> accumulatedValues = null;
+
+  /**
+   * Creates a new CFAbstractValue.
+   *
+   * @param analysis the analysis class this value belongs to
+   * @param annotations the annotations in this abstract value
+   * @param underlyingType the underlying (Java) type in this abstract value
+   */
+  protected AccumulationValue(
+      CFAbstractAnalysis<AccumulationValue, ?, ?> analysis,
+      AnnotationMirrorSet annotations,
+      TypeMirror underlyingType) {
+    super(analysis, annotations, underlyingType);
+    if (underlyingType.getKind() == TypeKind.TYPEVAR
+        || underlyingType.getKind() == TypeKind.WILDCARD) {
+      AccumulationAnnotatedTypeFactory typeFactory =
+          (AccumulationAnnotatedTypeFactory) analysis.getTypeFactory();
+      AnnotationMirror accumulator = null;
+      for (AnnotationMirror anm : annotations) {
+        if (typeFactory.isAccumulatorAnnotation(anm)) {
+          accumulator = anm;
+          break;
+        }
+      }
+      if (accumulator != null) {
+        accumulatedValues = new HashSet<>(typeFactory.getAccumulatedValues(accumulator));
+      }
+    }
+  }
+
+  /**
+   * If the underlying type is a type variable or a wildcard, then this is a set of accumulated
+   * values. Otherwise, it is null.
+   *
+   * @return the set (this is not a copy of the set, but an alias)
+   */
+  public Set<String> getAccumulatedValues() {
+    return accumulatedValues;
+  }
+
+  @Override
+  public AccumulationValue leastUpperBound(AccumulationValue other) {
+    AccumulationValue lub = super.leastUpperBound(other);
+    if (other == null || other.accumulatedValues == null || this.accumulatedValues == null) {
+      return lub;
+    }
+    // Lub the accumulatedValues by intersecting the lists as if they were sets.
+    lub.accumulatedValues = new HashSet<>(this.accumulatedValues.size());
+    lub.accumulatedValues.addAll(this.accumulatedValues);
+    lub.accumulatedValues.retainAll(other.accumulatedValues);
+    if (lub.accumulatedValues.isEmpty()) {
+      lub.accumulatedValues = null;
+    }
+    return lub;
+  }
+
+  @Override
+  public AccumulationValue mostSpecific(AccumulationValue other, AccumulationValue backup) {
+    AccumulationValue mostSpecific = super.mostSpecific(other, backup);
+    if (mostSpecific == null) {
+      if (other == null) {
+        return this;
+      }
+      // mostSpecific is null if the two types are not comparable.  This is normally
+      // because one of this or other is a type variable and annotations is empty, but the
+      // other annotations are not empty.  In this case, copy the accumulatedValues to the
+      // value with no annotations and return it as most specific.
+      if (other.getAnnotations().isEmpty()) {
+        other.addAccumulatedValues(this.accumulatedValues);
```

Comment:
This is returning greatest lower bound, but it should be returning:
if `this <: other`: return this;
if `other <: this`: return other;
otherwise: return backup;

Reply:
This code is based on very similar code in `KeyForValue`, which also seems (to me) to be computing a GLB. Is that code also incorrect?

Debugging this code on the failing test cases, I don't see that it's doing anything wrong. The interesting case is when `mostSpecific` is called on the following values:
```
this: CFAV{@CalledMethods("a"), T, [a] }
other: CFAV{, T, null }
```
In that case, the result is what I'd expect: `CFAV{, T, [a] }`

Would you expect something different here? Maybe I'm misunderstanding the specification of `mostSpecific`?

Reply:
I was wrong.  The problem was that values were added to `this` and `other` rather than created a new mostSpecific value.  I've made that change.